sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/action/QueryChangeStreamActionTest.java

```diff
@@ -148,10 +148,20 @@ public void testQueryChangeStreamWithDataChangeRecord() {
     when(changeStreamRecordMapper.toChangeStreamRecords(partition, rowAsStruct, resultSetMetadata))
         .thenReturn(Arrays.asList(record1, record2));
     when(dataChangeRecordAction.run(
-            partition, record1, restrictionTracker, outputReceiver, watermarkEstimator))
+            partition,
```

Comment:
Since their corresponding actions never get the instance of the `ThroughputEstimator` they won't be able to do that. However, I am adding tests that no update to the `ThroughputEstimator` will be done in `QueryChangeStreamAction`. It should cover the case you mentioned here.